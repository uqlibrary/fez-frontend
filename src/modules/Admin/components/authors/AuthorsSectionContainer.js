import React from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';
import { connect } from 'react-redux';
import { getFormValues } from 'redux-form/immutable';

import { FORM_NAME } from '../../constants';
import { FormValuesContext } from 'context';
import AuthorsSection from './AuthorsSection';
// import { NTRO_SUBTYPES, NTRO_SUBTYPE_CW_TEXTUAL_WORK } from 'config/general';
import { NTRO_SUBTYPES } from 'config/general';
export const AuthorsSectionContainer = ({ disabled, formValues, isNtro = false }) => {
    return (
        <FormValuesContext.Provider value={{ formValues: { ...formValues.toJS(), isNtro } }}>
            <AuthorsSection key={`disabled-${disabled}-isNtro-${isNtro}`} disabled={disabled} isNtro={isNtro} />
        </FormValuesContext.Provider>
    );
};

AuthorsSectionContainer.propTypes = {
    disabled: PropTypes.bool,
    formValues: PropTypes.object,
    isNtro: PropTypes.bool,
};

/* istanbul ignore next */
export const mapStateToProps = (state, ownProps) => {
    const formValues = getFormValues(FORM_NAME)(state) || Immutable.Map({});
    const subtype = formValues.get('adminSection')?.get('rek_subtype');
    return {
        disabled: ownProps.disabled,
        formValues: formValues.get('authorsSection') || Immutable.Map({}),
        isNtro: NTRO_SUBTYPES.includes(subtype), // && subtype !== NTRO_SUBTYPE_CW_TEXTUAL_WORK,
    };
};

export default connect(mapStateToProps)(React.memo(AuthorsSectionContainer));
