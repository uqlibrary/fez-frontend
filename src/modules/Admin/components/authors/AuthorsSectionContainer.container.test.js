import { AuthorsSectionContainer } from './AuthorsSectionContainer';

function setup(testProps = {}, args = { isShallow: true }) {
    const props = {
        formValues: {
            toJS: jest.fn(() => ({})),
        },
        ...testProps,
    };

    return renderComponent(AuthorsSectionContainer, props, args);
}

describe('AuthorsSectionContainer component', () => {
    it('should render default view', () => {
        const render = setup();
        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should render disabled view', () => {
        const render = setup({ disabled: true });
        expect(render.getRenderOutput()).toMatchSnapshot();
    });
});
