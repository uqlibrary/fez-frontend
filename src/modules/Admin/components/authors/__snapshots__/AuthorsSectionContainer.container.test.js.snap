// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsSectionContainer component should render default view 1`] = `
<Context.Provider
  value={
    {
      "formValues": {
        "isNtro": false,
      },
    }
  }
>
  <Memo(AuthorsSection)
    isNtro={false}
  />
</Context.Provider>
`;

exports[`AuthorsSectionContainer component should render disabled view 1`] = `
<Context.Provider
  value={
    {
      "formValues": {
        "isNtro": false,
      },
    }
  }
>
  <Memo(AuthorsSection)
    disabled={true}
    isNtro={false}
  />
</Context.Provider>
`;
